pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -326,3 +318,31 @@ func shouldProcessIngress(mesh *meshconfig.MeshConfig, ingress *v1beta1.Ingress)
 		return false
 	}
 }
+
+func createStringMatch(s string) *networking.StringMatch {
+	if s == "" {
+		return nil
+	}
+
+	// Replace "foo" with exact match
+	count := strings.Count(s, ".*")
+	if count == 0 {
+		return &networking.StringMatch{
+			MatchType: &networking.StringMatch_Exact{Exact: s},
+		}
+	}
+
+	// Replace "foo.*bar.*" and "foo.*bar" with regex match
+	if count > 1 || !strings.HasSuffix(s, ".*") {
```

Comment:
This is not the right way to check for regex strings. See proxy envoy v1 ingress.go. There is a go library thing that can check if a string is a regex or not.

Reply:
Actually can you just remove this entire piece ?